Source/Data/HACompressedEntity.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+public struct CompressedStatesUpdates: HADataDecodable {
+    public let add: [String: CompressedEntityState]?
+    public let remove: [String]?
+    public let change: [String: CompressedEntityDiff]?
```

Comment:
no need to do `let` - this is a struct

Reply:
A few reasons: it makes writing tests easier in some contexts, it allows you to mutate data easier (modify a copy rather than initialize an entirely new one), and it's already a language feature (you can't change the value anywhere anyway)